### PR TITLE
chore(asset.precision): replace magic 18

### LINKED ISSUE #1085: Abstract out magic precision number usages across transaction components
this `18` should probably be `feeAsset.precision` and abstracted out

_Originally posted by @0xdef1cafe in https://github.com/shapeshift/web/pull/1076#discussion_r816344975_

### PR DESCRIPTION
**Pull request by brymut**
## Description

Abstract out hardcoded magic 18 with appropriate abstracted out `asset.precision` property

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes #1085

### CODE REVIEW

**Comment by brymut** [comment]
Hi @0xApotheosis , I'd appreciate your opinion on this. I can't find any more hardcoded `18` usages across the Transaction components beyond these 2 occurences where they seem to be intentional fallbacks where they make sense:

https://github.com/shapeshift/web/blob/6a7ad73c6d69ac2c4ec7c840013e1d6f35699ffb/src/components/TransactionHistoryRows/utils.ts#L9-L12

https://github.com/shapeshift/web/blob/6a7ad73c6d69ac2c4ec7c840013e1d6f35699ffb/src/hooks/useTxDetails/useTxDetails.ts#L158

The rest of the occurrences happen in tests where there is no state to get the abstracted information from and hardcoded magic 18 makes sense too like to name a few:

https://github.com/shapeshift/web/blob/6a7ad73c6d69ac2c4ec7c840013e1d6f35699ffb/src/components/AssetSearch/helpers/filterAssetsBySearchTerm/filterAssetsBySearchTerm.test.ts#L1-L25

https://github.com/shapeshift/web/blob/6a7ad73c6d69ac2c4ec7c840013e1d6f35699ffb/src/test/mocks/assets.ts#L1-L25

Otherwise I think I've covered all the usages imo based on the issue description, unless I'm missing something

**Review by 0xApotheosis — DISMISSED** [review]
Thanks @brymut, it looks like the magic numbers were already removed internally since this issue was created, but this issue wasn't linked and closed.

**Comment by brymut** [comment]
ok, so what do we do?

**Review by gomesalexandre — APPROVED** [review]
Tested locally and `sellAsset.precision` has the value of the previous magic 18 number in the diffed branch. LGTM!

**Review by elmutt — APPROVED** [review]

**Review by elmutt — DISMISSED** [review]
Sell asset is the wrong preicion here for approval fee. For example if sellAsset is an erc20 that wont work. 18 works because right now we only use zrx but its still wrong

@majorhayes is correct we should be getting the fee asset

**Comment by brymut** [comment]
You're right @cjthompson @elmutt , I've switched it over.

File: src/components/Trade/hooks/useSwapper/useSwapper.ts [hunk 1]
```diff
@@ -237,7 +237,7 @@ export const useSwapper = () => {
       case 'eip155:1':
         {
           const approvalFee = bnOrZero(trade?.feeData?.chainSpecific?.approvalFee)
-            .dividedBy(bn(10).exponentiatedBy(18))
+            .dividedBy(bn(10).exponentiatedBy(feeAsset.precision))
```
Comment (theoboldfrazier) [inline src/components/Trade/hooks/useSwapper/useSwapper.ts:240]: Good find.

Here's a stylistic suggestion, as it tends to be a bit cleaner.

```suggestion
            .dividedBy(`1e+${feeAsset.precision}`))
```

**Review by theoboldfrazier — APPROVED** [review]
Left a non-blocking suggestion.

Tested it out and looks good.

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
> 

i actually prefer the bn solution - it'll get caught by the compiler if wrong. the string interpolation won't.

**Pull request merged**